php/libraries/NDB_BVL_Instrument.class.inc

```diff
@@ -2707,10 +2707,15 @@ abstract class NDB_BVL_Instrument extends NDB_Page
      */
     public function handle(ServerRequestInterface $request) : ResponseInterface
     {
-        if ($request->getMethod() === "POST") {
+        if ($request->getMethod() === "POST" && !isset($_POST['ClearInstrument'])) {
             $this->save();
         }
-
+        if (isset($_POST['ClearInstrument'])
+            && isset($_POST['deleteconfirm'])
```

Comment:
I agree with @ridz1208 that this is already getting called elsewhere and the solution shouldn't be to duplicate it enough.. the isset from the first `if` statement should be sufficient to fix the 500 bug.